File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterPagerAdapter.java
package ai.elimu.vitabu.ui.storybook;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.List;

import ai.elimu.model.enums.ReadingLevel;
import ai.elimu.model.v2.gson.content.StoryBookChapterGson;

public class ChapterPagerAdapter extends FragmentStatePagerAdapter {

    public static List<StoryBookChapterGson> storyBookChapters;

    private ReadingLevel readingLevel;

    private String description;

    public ChapterPagerAdapter(FragmentManager fm, List<StoryBookChapterGson> storyBookChapters, ReadingLevel readingLevel, String description) {
        super(fm);
        ChapterPagerAdapter.storyBookChapters = storyBookChapters;
        this.readingLevel = readingLevel;
        this.description = description;
    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0) {
            return CoverFragment.newInstance(readingLevel, description);
        } else {
            return ChapterFragment.newInstance(position, readingLevel);
        }
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return null;
    }

    @Override
    public int getCount() {
        return storyBookChapters.size();
    }
}

File: app/src/main/java/ai/elimu/vitabu/ui/storybook/ChapterPagerAdapter.kt
package ai.elimu.vitabu.ui.storybook

import ai.elimu.model.v2.enums.ReadingLevel
import ai.elimu.model.v2.gson.content.StoryBookChapterGson
import ai.elimu.vitabu.ui.storybook.ChapterFragment.Companion.newInstance
import ai.elimu.vitabu.ui.storybook.CoverFragment.Companion.newInstance
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class ChapterPagerAdapter(
    fm: FragmentManager,
    storyBookChapters: List<StoryBookChapterGson>,
    readingLevel: ReadingLevel,
    description: String
) :
    FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private val readingLevel: ReadingLevel

    private val description: String

    init {
        Companion.storyBookChapters = storyBookChapters
        this.readingLevel = readingLevel
        this.description = description
    }

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            newInstance(readingLevel, description)
        } else {
            newInstance(position, readingLevel)
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return null
    }

    override fun getCount(): Int {
        return storyBookChapters.size
    }

    companion object {
        var storyBookChapters: List<StoryBookChapterGson> = listOf()
    }
}
